### PR TITLE
Video Listing Page, Liked Videos & History

### PR DESCRIPTION
**Pull request by irajbhinde**
Hey! Please review my code & let me know if any changes are to be made

Functionalities Added -

1. Single Video Page 
2. Liked Videos Page
3. History Page

Files to check - 

- [ ] src/Utils/RequiresAuth.js
- [ ] src/pages/History Page/HistoryPage.jsx
- [ ] src/pages/Liked Videos/LikedVideosPage.jsx
- [ ] src/pages/Video Page/VideoPage.jsx
- [ ] src/Components/Cards/Cards.jsx - Video Player Card Line 222
- [ ] src/Components/Cards/Cards.jsx - History Page Card Line 322

GIFY - 

liked_vids_history2

### CODE REVIEW

File: src/Components/login_signup/LoginCard.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import { useState } from "react";
+import "../../pages/login/login-page.css";
+import { Link, useNavigate, useLocation } from "react-router-dom";
+import { useAuth } from "../context/auth-context";
+import { useFormik } from "formik";
```
Comment (harshalini) [inline src/Components/login_signup/LoginCard.jsx:5]: This is new to me, will explore this!

**Review by harshalini — APPROVED** [review]
The code is good, I liked the naming conventions

File: src/Components/Cards/cards.css [hunk 2]
```diff
@@ -92,15 +92,127 @@
...
+
+.video-features{
+  justify-content: space-between;
+  border-bottom: 2px solid var(--hover-graybg);
+  padding-bottom: 1rem;
+}
+
+.video-functions{
+  gap: 2rem !important; 
+}
```
Comment (ark2002) [inline src/Components/Cards/cards.css:199]: can avoid using important

File: src/Components/Playlist Modal/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+          </div>
+        ) : (
+          <div className="listOfPlaylist-container flex_c">
+            {playlist.map((item) => {
+              return (
+                <>
+                  <div className="modal-content flex_r">
+                    <input
+                      className="playlist-checkbox"
+                      checked={videoPresentInPlaylist(item, currentVideo)}
```
Comment (ark2002) [inline src/Components/Playlist Modal/Modal.jsx:49]: this is a nice implementation

File: src/Components/reducers/featureReducer.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    case "ADD_TO_LIKED_VIDEOS":
+      return {
+        ...state,
+        likedVideos: [...state.likedVideos, payload],
+      };
+    case "DELETE_FROM_LIKED_VIDEOS":
+      return {
+        ...state,
+        likedVideos: [
+          ...state.likedVideos.filter((vid) => vid._id !== payload._id),
```
Comment (ark2002) [inline src/Components/reducers/featureReducer.jsx:13]: dont need to spread if u are using filter

File: src/pages/Liked Videos/LikedVideosPage.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                >
+                  Explore
+                </button>
+              </div>
+            </>
+          ) : (
+            <div className="likedVideos">
+              {likedVideos.map((video) => {
+                return <VideoListingCard video={video} key={video._id} />;
+              })}
```
Comment (ark2002) [inline src/pages/Liked Videos/LikedVideosPage.jsx:37]: could use direct arrow function here

File: src/pages/Playlist Videos/PlaylistVideos.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            <div className="playlistVideos">
+              {playlistVideos.map((vids) => {
+                return (
+                  <PlaylistVideoCard
+                    playlistId={playlistId}
+                    video={vids}
+                    key={vids._id}
+                  />
+                );
+              })}
```
Comment (ark2002) [inline src/pages/Playlist Videos/PlaylistVideos.jsx:51]: same here

File: src/pages/Video Listing/video-listing-page.css [hunk 7]
```diff
@@ -24,3 +26,16 @@
     "sidebar video-main video-main"
     "sidebar video-main video-main";
 }
+
+.playlist-overlay {
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
```
Comment (ark2002) [inline src/pages/Video Listing/video-listing-page.css:35]: dont need this

File: src/pages/Watch Later Videos/WatchLaterVideos.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+                >
+                  Explore
+                </button>
+              </div>
+            </>
+          ) : (
+            <div className="watchLater">
+              {watchLaterVideos.map((video) => {
+                return <VideoListingCard video={video} key={video._id} />;
+              })}
```
Comment (ark2002) [inline src/pages/Watch Later Videos/WatchLaterVideos.jsx:38]: here aswell dont really need the explicit return keyword

**Review by ark2002 — APPROVED** [review]
code looks good , the file structure is great as well .

**Review by simran61 — APPROVED** [review]
Looks good....clean code

**Pull request merged**